src/ray/gcs/gcs_server/test/gcs_placement_group_manager_test.cc

```diff
@@ -166,6 +166,20 @@ class GcsPlacementGroupManagerTest : public ::testing::Test {
     EXPECT_TRUE(WaitForCondition(condition, 10 * 1000));
   }
 
+  void WaitUntilIoServiceDone() {
+    // In this test, io service is running in a different thread.
+    // That means it can have a thread safety issue, if the test thread
+    // touches private attributes while io service is accessing it.
+    // This method returns when there's no more task in the io service,
```

Comment:
What prevents new tasks from being added after this function runs?

Reply:
(fine if there's nothing atm, just trying to understand the nature of the fix)

Reply:
Hmm I guess theoretically new task can be added if the API post another task within a task. But this case doesn't exist now

Reply:
But this is the blocking call, and post ensures the insertion ordering, so normally it'd work

Reply:
Awesome